dwave/plugins/torch/autoencoder/losses/kl_divergence.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2025 D-Wave
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+from dimod import Sampler
+
+from dwave.plugins.torch.boltzmann_machine import AbstractBoltzmannMachine
+
+__all__ = ["pseudo_kl_divergence_loss"]
+
+binary_cross_entropy_with_logits = torch.nn.BCEWithLogitsLoss()
+
+
+def pseudo_kl_divergence_loss(
+    spins: torch.Tensor,
+    logits: torch.Tensor,
+    boltzmann_machine: AbstractBoltzmannMachine,
+    sampler: Sampler,
+    sampler_kwargs: dict,
+):
+    """
+    A pseudo Kullback-Leibler divergence loss function for a Boltzmann machine. This is
+    not the true KL divergence, but the gradient of this function is the same as the KL
+    divergence gradient.
+
+    Args:
+        spins (torch.Tensor): A tensor of spins of shape (batch_size, n_spins) or shape
+            (batch_size, n_samples, n_spins) obtained from a stochastic function that
+            maps the output of the encoder (logit representation) to a spin
+            representation.
+        logits (torch.Tensor): A tensor of logits of shape (batch_size, n_spins). These
+            logits are the raw output of the encoder.
+        boltzmann_machine (AbstractBoltzmannMachine): An instance of a Boltzmann
+            machine.
+        sampler (Sampler): A sampler used for generating samples.
+        sampler_kwargs (dict): Additional keyword arguments for the ``sampler.sample``
+            method.
+
+    Returns:
+        torch.Tensor: The computed pseudo KL divergence loss.
+    """
+    samples = boltzmann_machine.sample(
+        sampler=sampler, device=spins.device, **sampler_kwargs
+    )
+    probabilities = torch.sigmoid(logits)
+    entropy_of_encoder = binary_cross_entropy_with_logits(logits, probabilities)
+    cross_entropy = torch.mean(boltzmann_machine(spins)) - torch.mean(
```

Comment:
use existing code, i.e., `boltzmann_machine.objective`.
should it be a separate function? 
cc @thisac

Reply:
Agreed. Seems like this should be
```python
    cross_entropy = boltzmann_machine.objective(spins, samples)
```

Reply:
Ooops, I meant to ask whether this should be an independent function instead of a method?
e.g., `objective(boltzmann_machine, spins_1, spins_2)` instead?

Reply:
> Agreed. Seems like this should be

Ooops, I meant to ask whether this should be an independent function instead of a method?
e.g., `objective(boltzmann_machine, spins_1, spins_2)` instead?

Reply:
Done! Didn't see this method existed. I think it's ok that this is a method.